packages/aws-cdk/test/cli/telemetry/file-sink.test.ts

```diff
@@ -0,0 +1,186 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { IoHelper } from '../../../lib/api-private';
+import { CliIoHost } from '../../../lib/cli/io-host';
+import { FileTelemetrySink } from '../../../lib/cli/telemetry/file-sink';
+import type { TelemetrySchema } from '../../../lib/cli/telemetry/schema';
+
+describe('FileTelemetrySink', () => {
```

Comment:
I think you're missing a test to validate that we throw if the file already exists.